src/components/Search.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState } from 'react';
+
+type UserType = {
+  id: string;
+  name: string;
+  login: string;
+  imgUrl: string; // avatar_url
+  description: string; // bio
+};
+
+type UserProps = {
+  user: UserType | null;
+};
+
+function Search({ user }: UserProps): React.ReactElement {
+  const [value, setValue] = useState('');
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => setValue(e.target.value);
+
+  const handleClick = () => {
+    // fetchUser();
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    e.preventDefault();
+    if (e.key === 'Enter') {
+      // fetchUser();
+    }
+  };
+
+  return (
+    <section className="absolute top-8 left-10 rounded-lg bg-white px-6  pt-2 pb-4 shadow-lg first-letter:space-y-4">
+      <form className="flex items-center space-x-2">
+        <input
+          className="border-b border-gray-200 px-1 py-2 focus:outline-none focus:ring-0 placeholder:text-base"
+          placeholder="Github 아이디 검색"
+          value={value}
+          onChange={handleChange}
+          onKeyDown={handleKeyDown}
+        />
+        <button type="button" className="text-gray-500" onClick={handleClick}>
```

Comment:
`type`을 기본값인 `submit`으로 하면 버튼 요소를 누를 때와 엔터키를 누를 때 둘 다 `form`의 `onSubmit` 이벤트로 발생하지 않을까요? 굳이 두 이벤트 콜백을 정의할 필요 없이요.

```suggestion
      <form className="flex items-center space-x-2" onSubmit={...}>
        <input
          className="border-b border-gray-200 px-1 py-2 focus:outline-none focus:ring-0 placeholder:text-base"
          placeholder="Github 아이디 검색"
          value={value}
          onChange={handleChange}
        />
        <button type="submit" className="text-gray-500">
```

Reply:
아 그렇네요..! 감사합니다